File: 240322/특정 수와 근접한 합/sum-close-to-particular-number.py
import sys

N, S = map(int,sys.stdin.readline().split())
nums = list(map(int,sys.stdin.readline().split()))

base = sum(nums)
ans = sys.maxsize

for i in range(N):
    for j in range(i+1,N):
        temp = base-nums[i]-nums[j]
        ans = min(ans,abs(temp-S))

print(ans)

File: 240404/특정 수와 근접한 합/sum-close-to-particular-number.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main() {
    int N, S, nums[100], base = 0, temp, diff, ans = INT_MAX;

    scanf("%d %d",&N,&S);
    for(int i = 0; i < N; i++) {
        scanf("%d",&nums[i]);
        base += nums[i];
    }
    
    for(int i = 0; i < N; i++)
        for(int j = i+1; j < N; j++) {
            temp = base - nums[i] - nums[j];
            diff = abs(S-temp);
            if(diff < ans)
                ans = diff;
        }
    
    printf("%d\n", ans);

    return 0;
}